napari/_qt/dialogs/preferences_dialog.py

```diff
@@ -260,6 +272,9 @@ def reject(self):
 
 def update_widget_state(name, widget):
     def _update_widget_state(event):
-        widget.state = {name: event.value}
+        value = event.value
+        if isinstance(value, StrEnum):
```

Comment:
So, I used here `StrEnum` since I only found an issue with the widgets that were using values like:

https://github.com/napari/napari/blob/f0c02a844cdd0eebceb64b696bb4ab73c2aed08f/napari/settings/_constants.py#L7

and

https://github.com/napari/napari/blob/f0c02a844cdd0eebceb64b696bb4ab73c2aed08f/napari/settings/_constants.py#L39

Which are based on `StrEnum` 

https://github.com/napari/napari/blob/f0c02a844cdd0eebceb64b696bb4ab73c2aed08f/napari/utils/compat.py#L5-L12

The main issue comes I think from `StrEnum` based classes (at least on my setup) returning when being cast as a string something like `<Class.Attribute>` instead of the actual enum value. Particulary for the preference dialog that was happening at

https://github.com/napari/napari/blob/f0c02a844cdd0eebceb64b696bb4ab73c2aed08f/napari/_vendor/qt_json_builder/qt_jsonschema_form/widgets.py#L788-L795

However, is true that maybe ensuring to always use the actual enum `value` for any `Enum` based class could be worthy 🤔 Let me know if I should change the validation from using `StrEnum` to the more general `Enum`!